codalab/rest/schemas.py

```diff
@@ -225,23 +225,23 @@ class BundleStoreSchema(Schema):
     id = fields.String(validate=validate_uuid, attribute='uuid')
     uuid = fields.String(attribute='uuid')
     owner = fields.Integer(attribute='owner_id')
-    name = fields.String()
-    storage_type = fields.String()
-    storage_format = fields.String()
+    name = fields.String(required=True)
+    storage_type = fields.String(required=True)
+    storage_format = fields.String(allow_none=True)
     url = fields.String(allow_none=True)
-    authentication = fields.String()
-    authentication_env = fields.String()
+    authentication = fields.String(allow_none=True)
+    authentication_env = fields.String(allow_none=True)
 
     class Meta:
-        type_ = 'bundle-store'
+        type_ = 'bundle_stores'
```

Comment:
The type name must match the endpoint URL (`/bundle_stores`) as per how our json api client works